File: library/src/main/java/com/pengrad/telegrambot/model/request/InlineQueryResultCachedMpeg4Gif.java
package com.pengrad.telegrambot.model.request;

import java.io.Serializable;

/**
 * Stas Parshin
 * 06 May 2016
 */
public class InlineQueryResultCachedMpeg4Gif extends InlineQueryResult<InlineQueryResultCachedMpeg4Gif> implements Serializable {
    private final static long serialVersionUID = 0L;

    private String mpeg4_file_id;

    private String title;
    private String caption;
    private String parse_mode;

    public InlineQueryResultCachedMpeg4Gif(String id, String fileId) {
        super("mpeg4_gif", id);
        this.mpeg4_file_id = fileId;
    }

    public InlineQueryResultCachedMpeg4Gif title(String title) {
        this.title = title;
        return this;
    }

    public InlineQueryResultCachedMpeg4Gif caption(String caption) {
        this.caption = caption;
        return this;
    }

    public InlineQueryResultCachedMpeg4Gif parseMode(ParseMode parseMode) {
        this.parse_mode = parseMode.name();
        return this;
    }
}

File: core/src/main/kotlin/com/pengrad/telegrambot/model/request/InlineQueryResultCachedMpeg4Gif.kt
package com.pengrad.telegrambot.model.request

import java.io.Serializable

/**
 * Stas Parshin
 * 06 May 2016
 */
class InlineQueryResultCachedMpeg4Gif(id: String, private val mpeg4_file_id: String) : InlineQueryResult<InlineQueryResultCachedMpeg4Gif>("mpeg4_gif", id), Serializable {

    private var title: String? = null
    private var caption: String? = null
    private var parse_mode: String? = null

    fun title(title: String): InlineQueryResultCachedMpeg4Gif {
        this.title = title
        return this
    }

    fun caption(caption: String): InlineQueryResultCachedMpeg4Gif {
        this.caption = caption
        return this
    }

    fun parseMode(parseMode: ParseMode): InlineQueryResultCachedMpeg4Gif {
        this.parse_mode = parseMode.name
        return this
    }

    companion object {
        private const val serialVersionUID = 0L
    }
}
